lib/scsi_defs.h

```diff
@@ -0,0 +1,121 @@
+#ifndef LIB_SCSI_DEFS_H
+#define LIB_SCSI_DEFS_H
+
+namespace scsi_defs {
+
+  enum Status {
```

Comment:
Might be worth explicitly enforcing the enums as one byte (enum Status : char {})

Reply:
enforcing the size is a good idea! What are your thoughts on using uint8_t instead of char

Reply:
Good point, I think uint8_t is better in this situation